src/riak_kv_w1c_worker.erl

```diff
@@ -108,7 +109,10 @@ put(RObj, Options) ->
             RObj2 = riak_object:set_vclock(RObj, vclock:fresh(<<0:8>>, 1)),
             RObj3 = riak_object:update_last_modified(RObj2),
             RObj4 = riak_object:apply_updates(RObj3),
-            EncodedVal = EncodeFn(RObj4),
+            MD  = riak_object:get_metadata(RObj4),
+            MD1 = dict:erase(?MD_TS_LOCAL_KEY, MD),
+            RObj5 = riak_object:update_metadata(RObj4, MD1),
```

Comment:
From what I understand reading the `riak_object:apply_updates` code (https://github.com/basho/riak_kv/blob/develop/src/riak_object.erl#L540), it needs to be the final step here, sealing the changes introduced in `riak_object:update_metadata`.